presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp

```diff
@@ -148,12 +182,41 @@ sendGet(http::HttpClient* client, const std::string& url, MemoryPool* pool) {
       .url(url)
       .send(client, pool);
 }
+} // namespace
+
+class HttpTest : public ::testing::Test {};
 
-TEST(HttpTest, basic) {
+TEST_F(HttpTest, ssl) {
   auto memoryPool = getProcessDefaultMemoryManager().getPool(
       "basic", MemoryPool::Kind::kLeaf);
+
+  std::string certPath = getCertsPath("test_cert1.pem");
+  std::string keyPath = getCertsPath("test_key1.pem");
+  std::string ciphers = "AES128-SHA,AES128-SHA256,AES256-GCM-SHA384";
+
+  auto httpsConfig = std::make_unique<http::HttpsConfig>(
+      folly::SocketAddress("127.0.0.1", 0), certPath, keyPath, ciphers);
+
   auto server =
-      std::make_unique<http::HttpServer>(folly::SocketAddress("127.0.0.1", 0));
+      std::make_unique<http::HttpServer>(nullptr, std::move(httpsConfig));
+
+  HttpServerWrapper wrapper(std::move(server));
+  auto serverAddress = wrapper.start().get();
+
+  folly::EventBase evb;
+  auto ctx = std::make_shared<folly::SSLContext>();
+  folly::AsyncSSLSocket::UniquePtr sock(new folly::AsyncSSLSocket(ctx, &evb));
+  Cb cb(sock.get());
+  sock->connect(&cb, serverAddress, 1000);
+  evb.loop();
+  EXPECT_TRUE(cb.success);
+}
+
+TEST_F(HttpTest, basic) {
```

Comment:
Can we use TEST_P to test both setup for each test case? Thanks!

Reply:
I will do that in the next PR. This PR only refactors server. Other tests in this file requires HttpClient refactoring, which I plan to do next and run all tests for both mode.